crates/rooch/src/commands/account/commands/transfer.rs

```diff
@@ -0,0 +1,138 @@
+// Copyright (c) RoochNetwork
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::cli_types::{CommandAction, TransactionOptions, WalletContextOptions};
+use async_trait::async_trait;
+use clap::Parser;
+use move_command_line_common::types::ParsedStructType;
+use move_core_types::u256::U256;
+use rooch_key::key_derive::verify_password;
+use rooch_key::keystore::account_keystore::AccountKeystore;
+use rooch_rpc_api::jsonrpc_types::ExecuteTransactionResponseView;
+use rooch_types::address::{MultiChainAddress, RoochAddress};
+use rooch_types::framework::transfer::TransferModule;
+use rooch_types::multichain_id::{self, RoochMultiChainID};
+use rooch_types::{
+    error::{RoochError, RoochResult},
+    transaction::rooch::RoochTransaction,
+};
+use rpassword::prompt_password;
+
+/// Transfer coin
+#[derive(Debug, Parser)]
+pub struct TransferCommand {
+    /// The account's address to receive coin, if account not exist, auto create account.
+    #[clap(short = 't', long)]
+    to: String,
+
+    /// Struct name as `<ADDRESS>::<MODULE_ID>::<STRUCT_NAME><TypeParam>`
+    /// Example: `0x3::gas_coin::GasCoin`, `0x123::Coin::Box<0x123::coin_box::FCoin>`
+    #[clap(short = 'c', long, value_parser=ParsedStructType::parse)]
+    coin_type: ParsedStructType,
+
+    /// The amount of coin to transfer
+    #[clap(short = 'a', long)]
+    amount: U256,
+
+    /// chain_id  [ROOCH: 20230101] [BITCOIN: 0] [ETHER: 60]
+    #[clap(long, default_value_t = multichain_id::ROOCH)]
+    chain_id: u64,
```

Comment:
Should we use the `multi_chain_id` here? avoid to conflict with `chain_id(local/dev/test/main)`

Reply:
Or, this command only supports rooch addresses. If we want to support Bitcoin, we should provide a new command that accepts the Bitcoin address as an argument.

Reply:
I guess it will generate a different rooch address from bitcoin or eth address. But I don't know what the rooch address is. The `balance` command only supports rooch addresses.

I did tests on my local server.
For example, to transfer coin on the Bitcoin chain:
`rooch account transfer --to bc1qxy2kgdygjrsqtzq2n0yrf2493p83kkfjhx0wlh -c 0xe952d2f26b773a791ef03604bd323184ec2587d9dd67297c1c598f8252d35256::fixed_supply_coin::FSC -a 1000000 --chain-id 0`
It was successful, because when I run `rooch account balance`, its balance is correctly reduced by 1000000.

But, it can't get the receiver's balance.
```
rooch account balance -a bc1qxy2kgdygjrsqtzq2n0yrf2493p83kkfjhx0wlh
Error: Unbound named address: 'bc1qxy2kgdygjrsqtzq2n0yrf2493p83kkfjhx0wlh'
```

When I conduct tests on the Ethereum chain, the `balance` command incorrectly identifies the Ethereum address as a rooch address.

Reply:
I think there is an error.
Here is the transfer logic in `transfer.move`.
```
public entry fun transfer_coin_to_multichain_address<CoinType: key + store>(

from: &signer,
multichain_id: u64,
raw_address: vector<u8>,
amount: u256,
) {
let maddress = multichain_address::new(multichain_id, raw_address);
let to = address_mapping::resolve(maddress);
assert!(option::is_some(&to), ErrorAccountNotExists);

let to = option::extract(&mut to);
if(!account::exists_at(to)) {
account_entry::create_account(to);
};
account_coin_store::transfer<CoinType>(from, to, amount)
}
```

 I'd like to change this.
```
-let to = address_mapping::resolve(maddress);
+let to = address_mapping::resolve_or_generate(maddress);
```
This whill generate a new rooch address using black2b256 based on the Multichainaddress.

Reply:
But I don't know why the original code works, so I didn't change the code.

Reply:
> Should we use the `multi_chain_id` here? avoid to conflict with `chain_id(local/dev/test/main)`

There won't be a conflict.

Reply:
The account create logic changed in #1624

We could auto-detect the chain ID via the address string. The address string formats are different.

Reply:
We can remove auto-create for account, in the future, we need to deprecated the `CoinStores` in account resource, and then we can support transfer coin to no exists account.